.az/ios_packaging.yml

```diff
@@ -0,0 +1,107 @@
+# packaging pipeline for iOS CocoaPods package
+
+parameters:
+- name: IsReleaseBuild
+  displayName: "Is this is a release build?"
+  type: boolean
+  default: false
+
+name: "$(Date:yyyyMMdd)$(Rev:rrr)"  # build number format
+
+jobs:
+- job: IosPackaging
+  displayName: "iOS Packaging"
+
+  pool:
+    vmImage: "macOS-12"
+
+  timeoutInMinutes: 120
+
+  steps:
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: "3.9"
+      addToPath: true
+      architecture: "x64"
+
+  # iOS xcframework build doesn't work with CMake 3.25.1, pin to 3.25.0
+  - script: |
+      python -m pip install cmake==3.25.0
+    displayName: "Install CMake 3.25.0"
+
+  - script: |
+      python ./tools/ios/build_xcframework.py \
+        --output-dir $(Build.BinariesDirectory)/xcframework_out \
+        --config Release
+    displayName: "Build xcframework"
+
+  # <repo root>/.version.txt is generated by a build, so we put the step that reads it after the build step
+  - bash: |
+      set -e -x
+      BASE_VERSION="$(cat ./.version.txt)"
+
+      if [[ "${{ parameters.IsReleaseBuild }}" == "true" ]]; then
+        VERSION="${BASE_VERSION}"
+      else
+        SHORT_COMMIT_HASH="$(git rev-parse --short HEAD)"
+        VERSION="${BASE_VERSION}-dev+$(Build.BuildNumber).${SHORT_COMMIT_HASH}"
+      fi
+
+      set_var() {
+        local VAR_NAME=${1:?}
+        local VAR_VALUE=${2:?}
+        echo "##vso[task.setvariable variable=${VAR_NAME}]${VAR_VALUE}"
+        echo "${VAR_NAME}: ${VAR_VALUE}"
+      }
+
+      set_var "ORT_EXTENSIONS_POD_VERSION" "${VERSION}"
+    displayName: "Set ORT_EXTENSIONS_POD_VERSION variable"
```

Comment:
not a comment, question:

noticed this is added as an extra step compared to the `CI pipline - IosPackage job`, do we have to do similar set environment variable thing for Android packaging pipline?

Reply:
ah sorry, I think I see. In the CI pipline, it is using some dev version pods, here we just formalize it.

Reply:
yeah, this is just adjusting the version based on the IsReleaseBuild parameter. maybe I can put it into a template so we can reuse it.
would be good to have something similar for the Android pipeline.